wsmaster/che-core-api-factory-azure-devops/src/main/java/org/eclipse/che/api/factory/server/azure/devops/AzureDevOpsApiClient.java

```diff
@@ -0,0 +1,164 @@
+      @Named("che.integration.azure.devops.application_scopes") String[] scopes) {
+    this.scopes = scopes;
+    this.azureDevOpsApiEndpoint = trimEnd(azureDevOpsApiEndpoint, '/');
+    this.httpClient =
+        HttpClient.newBuilder()
+            .executor(
+                Executors.newCachedThreadPool(
+                    new ThreadFactoryBuilder()
+                        .setUncaughtExceptionHandler(LoggingUncaughtExceptionHandler.getInstance())
+                        .setNameFormat(AzureDevOpsApiClient.class.getName() + "-%d")
+                        .setDaemon(true)
+                        .build()))
+            .connectTimeout(DEFAULT_HTTP_TIMEOUT)
+            .version(HttpClient.Version.HTTP_1_1)
+            .build();
+  }
+
+  /**
+   * Returns the user associated with the provided OAuth access token. See Microsoft documentation
+   * at:
+   *
+   * <p>https://learn.microsoft.com/en-us/rest/api/azure/devops/graph/users/get?view=azure-devops-rest-7.0&tabs=HTTP
+   */
+  public AzureDevOpsUser getUserWithOAuthToken(String token)
+      throws ScmItemNotFoundException, ScmCommunicationException, ScmBadRequestException {
+    final String url =
+        String.format(
+            "%s/_apis/profile/profiles/me?api-version=%s",
+            azureDevOpsApiEndpoint, AzureDevOps.API_VERSION);
+    return getUser(url, formatAuthorizationHeader(token, false));
+  }
+
+  /**
+   * Returns the user associated with the provided PAT. The difference from {@code
+   * getUserWithOAuthToken} is in authorization header and the fact that PAT is associated with
+   * organization.
+   */
+  public AzureDevOpsUser getUserWithPAT(String pat, String organization)
+      throws ScmItemNotFoundException, ScmCommunicationException, ScmBadRequestException {
+    final String url =
+        String.format(
+            "%s/%s/_apis/profile/profiles/me?api-version=%s",
+            azureDevOpsApiEndpoint, organization, AzureDevOps.API_VERSION);
+    return getUser(url, formatAuthorizationHeader(pat, true));
+  }
+
+  private AzureDevOpsUser getUser(String url, String authorizationHeader)
+      throws ScmItemNotFoundException, ScmCommunicationException, ScmBadRequestException {
+    final HttpRequest userDataRequest =
+        HttpRequest.newBuilder(URI.create(url))
+            .headers("Authorization", authorizationHeader)
+            .timeout(DEFAULT_HTTP_TIMEOUT)
+            .build();
+
+    LOG.trace("executeRequest={}", userDataRequest);
+    return executeRequest(
+        httpClient,
+        userDataRequest,
+        response -> {
+          try {
+            return OBJECT_MAPPER.readValue(response.body(), AzureDevOpsUser.class);
+          } catch (IOException e) {
+            throw new UncheckedIOException(e);
+          }
+        });
+  }
+
+  /** Returns the scopes of the OAuth token. */
+  public String[] getTokenScopes(String authenticationToken)
```

Comment:
The `authenticationToken` parameter is redundant

Reply:
Let's keep it for the case I found the way to get the scopes.

Reply:
Please update the javadoc then.